doc/TRAINING.rst

```diff
@@ -196,6 +196,21 @@ python3 DeepSpeech.py --train_files ./train.csv --dev_files ./dev.csv --test_fil
 
 On a Volta generation V100 GPU, automatic mixed precision speeds up DeepSpeech training and evaluation by ~30%-40%.
 
+Distributed training using Horovod
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you have a capable compute architecture, it is possible to distribute the training using `Horovod <https://github.com/horovod/horovod>`_. A fast network is recommended.
+Horovod is capable of using MPI and NVIDIA's NCCL for highly optimized inter-process communication.
+It also offers `Gloo <https://github.com/facebookincubator/gloo>`_ as an easy-to-setup communication backend.
+
+For more information about setup or tuning of Horovod please visit `Horovod's Github <https://github.com/horovod/horovod>`_.
+
+To train on 4 machines using 4 GPUs each:
```

Comment:
Also, are there any requirements:
 - same GPUs ?
 - same OS ?
 - same drivers ?
 - same number of GPUs on each system ?

Reply:
However, Horovod is expected to run on highly heterogeneous systems, it is not documented well what works. 
I would not support something other than different GPUs per machine.This can be controlled by the number of processes per host, since every process has only one GPU pinned to it.
`horovodrun -np 16 -H server1:NUMPROC_1,server2:NUMPROC_2,server3 [...]`

Reply:
Maybe you can add link to Horovod doc covering those aspects ? And articulate that we cannot support anything besides homogenous configurations and that heterogenous cases will have to deal with their mess ? :)